main.tf

```diff
@@ -16,9 +20,11 @@ resource "aws_iam_openid_connect_provider" "chainguard_idp" {
 }
 
 resource "aws_iam_role" "canary_role" {
+  for_each = toset(local.enforce_group_ids)
+
   // Canary role has no permissions, but is used to validate that AWS account
   // connection has been correctly set up.
-  name = "chainguard-canary"
+  name = "chainguard-canary-${each.value}"
```

Comment:
Have you checked that this works? IIRC we bind to the IAM role by name and the name must be exactly `chainguard-canary` exactly. If we want many groups to be able to account associate, we need one role, but to expand the trust relationship to include a subject for each Enforce IAM GROUP.

Reply:
fixed.